baybe/searchspace/core.py

```diff
@@ -304,22 +304,23 @@ def estimate_product_space_size(parameters: Iterable[Parameter]) -> MemorySize:
     def transform(
         self,
         data: pd.DataFrame,
+        allow_missing: bool = False,
     ) -> pd.DataFrame:
-        """Transform data from experimental to computational representation.
-
-        This function can e.g. be used to transform data obtained from measurements.
-        Continuous parameters are not transformed but included.
+        """Transform parameters from experimental to computational representation.
```

Comment:
I think this is partly due to legacy naming. Just to be on the same page about what the method does: it transforms `the parameter-related columns` of a dataframe from `exp_rep` to `comp_rep`, but does **not** touch any other columns (e.g. target columns or randomly added columns). So writing `data` would be confusing, since it only affects the parameters.

That said, we can of course if it's worth adjusting a few things. One **could** rename the input argument (e.g. plain `df` or something more suggestive) or we **could** require that only parameter columns are present... But I am not sure if that's worth a deprecation / if it's even desirable. Just pointing out these ideas since apparently the current version is not clear enough!?

@Scienfitz, @AVHopp?

Reply:
Gave it another thought: how about we take this as an opportunity and make the behavior of the `transform` functions – which is a bit non-obvious at the moment – perfectly precise/robust?

What I could offer is to turn in into:
```python
    def transform(
        self,
        df: pd.DataFrame,
        *,
        allow_missing: bool = False,
        allow_additional: bool =False,
    ) -> pd.DataFrame:
```
which by default uses the most strict setting, to avoid user mistakes.

The whole thing could be made completely backward-compatible via deprecation using a temporary interface like
```python
    def transform(
        self,
        df: pd.DataFrame | None,
        *,
        allow_missing: bool = False,
        allow_additional: bool = bool | None,
        data: pd.DataFrame | None,
    ) -> pd.DataFrame:
```
which would exactly recover the original behavior if people used the old `data` argument or leave the `allow_additional` flag unspecified

Reply:
Like the idea, together with a corresponding description à la "Transform the parameter-related columns of df..." that would be gucci. Would however be interested in @Scienfitz opinion here.

Reply:
Have created a separate PR for this: #289 
Once approved and merged, I can rebase this one on top, which will resolve the thread

Reply:
like `df`, no deprecation needed

Reply:
All done and rebased